### PR TITLE
UnsubscribePage: tell CSRFDetector that we check the token here

### PR DESCRIPTION
**Pull request by macbre**
[PLATFORM-2389](https://wikia-inc.atlassian.net/browse/PLATFORM-2389)

And use `hash_equals()` to validate a token.

@Grunny

### CODE REVIEW

File: extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php [hunk 1]
```diff
@@ -74,12 +77,16 @@ public function execute( $subpage ) {
...
 			// $wgOut->addHTML("shouldtoken={$shouldToken}\n"); #DEVL (remove before release)
 			return;
 		}
 
+		// PLATFORM-2389: request token is checked above
+		// tell CSRFDetector that we're fine here
+		CSRFDetector::onUserMatchEditToken();
+
 		#does the non-blank email they gave us look like an email?
 		if( Sanitizer::validateEmail( $email ) == false ) {
```
Comment (WikiaTech) [inline extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php:91]: MINOR Remove the literal "false" boolean value. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 18 issues
* MAJOR 13 major
* MINOR 2 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Unsubscribe.body.php#L37](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L37): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MAJOR [Unsubscribe.body.php#L38](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L38): Remove this unused "$wgUser" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. MAJOR [Unsubscribe.body.php#L46](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L46): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [Unsubscribe.body.php#L145](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L145): Remove this unused "$wgLang" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. MAJOR [Unsubscribe.body.php#L145](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L145): Remove this unused "$wgScript" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. MAJOR [Unsubscribe.body.php#L170](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L170): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [Unsubscribe.body.php#L189](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L189): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [Unsubscribe.body.php#L215](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L215): Remove this unused "$form_action" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. MAJOR [Unsubscribe.body.php#L216](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L216): Remove this unused "$form_title" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. MAJOR [Unsubscribe.body.php#L218](https://github.com/Wikia/app/blob/88df28c2b702d1eb0643586c6d93c75ad2e35a9d/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L218): Remove this unused "$unsub_legend" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php [hunk 2]
```diff
@@ -32,7 +35,7 @@ function getDescription() {
 	 * @param $subpage Mixed: parameter passed to the page or null
 	 */
 	public function execute( $subpage ) {
```
Comment (WikiaTech) [inline extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php:37]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
- MAJOR 10 major
- MINOR 2 minor
- INFO 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [Unsubscribe.body.php#L46](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L46): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
2. MAJOR [Unsubscribe.body.php#L170](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L170): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
3. MAJOR [Unsubscribe.body.php#L189](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L189): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
4. MAJOR [Unsubscribe.body.php#L215](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L215): Remove this unused "$form_action" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
5. MAJOR [Unsubscribe.body.php#L216](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L216): Remove this unused "$form_title" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
6. MAJOR [Unsubscribe.body.php#L218](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L218): Remove this unused "$unsub_legend" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
7. MAJOR [Unsubscribe.body.php#L219](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L219): Remove this unused "$unsub_text" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
8. MAJOR [Unsubscribe.body.php#L220](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L220): Remove this unused "$unsub_button" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
9. MAJOR [Unsubscribe.body.php#L257](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L257): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
10. MINOR [Unsubscribe.body.php#L51](https://github.com/Wikia/app/blob/79ee82b950a19e790256f7c3601ae270eb52aeee/extensions/wikia/SpecialUnsubscribe/Unsubscribe.body.php#L51): Define a constant instead of duplicating this literal "token" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by Grunny** [comment]
LGTM.

**Pull request merged**